inspec/a2-api-integration/controls/iam_v2.rb

```diff
@@ -750,4 +750,285 @@
+    end
+
+    after(:all) do
+      resp = automate_api_request("/apis/iam/v2beta/projects/#{CUSTOM_PROJECT_ID}", http_method: 'DELETE')
+      expect(resp.http_status.to_s).to match(/200|404/)
+    end
+
+    context "when the project does not exist" do
+      before(:each) do
+        resp = automate_api_request("/apis/iam/v2beta/projects/#{CUSTOM_PROJECT_ID}", http_method: 'DELETE')
+        expect(resp.http_status.to_s).to match(/200|404/)
+      end
+
+      after(:all) do
+        resp = automate_api_request("/apis/iam/v2beta/projects",
+          http_method: 'POST',
+          request_body: {
+            id: CUSTOM_PROJECT_ID,
+            name: "display name !#$#"
+          }.to_json
+        )
+        expect(resp.http_status).to eq 200
+      end
+
+      describe "GET /iam/v2beta/projects/:id/rules" do
+        it "returns an empty list" do
+          resp = automate_api_request("/apis/iam/v2beta/projects/project-not-found/rules")
+          expect(resp.http_status).to eq 200
+          expect(resp.parsed_response_body[:rules]).to eq([])
+        end
+      end
+
+      describe "GET /iam/v2beta/rules" do
+        it "returns an empty list" do
+          resp = automate_api_request("/apis/iam/v2beta/rules")
+          expect(resp.http_status).to eq 200
+          expect(resp.parsed_response_body[:rules]).to eq([])
+        end
+      end
+    end
+
+    context "when there are no rules but the project exists" do
+      describe "GET /iam/v2beta/projects/:id/rules" do
+        it "returns an empty list" do
+          resp = automate_api_request("/apis/iam/v2beta/projects/#{CUSTOM_PROJECT_ID}/rules")
+          expect(resp.http_status).to eq 200
+          expect(resp.parsed_response_body[:rules]).to eq([])
+        end
+      end
+
+      describe "GET /iam/v2beta/rules" do
+        it "returns an empty list" do
+          resp = automate_api_request("/apis/iam/v2beta/rules")
+          expect(resp.http_status).to eq 200
+          expect(resp.parsed_response_body[:rules]).to eq([])
+        end
+      end
+
+      describe "GET /iam/v2beta/rules/:id" do
+        it "returns a 404" do
+          resp = automate_api_request("/apis/iam/v2beta/rules/not-found")
+          expect(resp.http_status).to eq 404
+          expect(resp.parsed_response_body[:rule]).to eq(nil)
+        end
+      end
+
+      describe "POST /iam/v2beta/rules" do
+        CUSTOM_RULE = {
+          id: "custom-rule",
+          name: "display name !#$#",
+          project_id: CUSTOM_PROJECT_ID,
+          type: "NODE",
+          conditions: [
+            {
+              type: "CHEF_TAGS",
+              operator: "MEMBER_OF",
+              values: ["tag1", "tag2"]
+            }
+          ]
+        }
+
+        after(:each) do
+          resp = automate_api_request("/apis/iam/v2beta/rules/#{CUSTOM_RULE[:id]}", http_method: 'DELETE')
+          expect(resp.http_status.to_s).to match(/200|404/)
+        end
+
+        it "creates a new rule" do
```

Comment:
These, for example, are organized in a way that makes a little more sense to me. 

Perhaps we have a describe block for the general api ("rules") and then each test is like "creates the rule with correct params" or "updates the rule" or "update fails with 404 when rule is not found"